README.md

```diff
@@ -2,12 +2,25 @@
 
 **Distributed delayed jobs in nodejs**. Resque is a background job system backed by [Redis](http://redis.io) (version 2.6.0 and up required). It includes priority queues, plugins, locking, delayed jobs, and more! This project is a very opinionated but API-compatible with [Resque](https://github.com/resque/resque) and [Sidekiq](http://sidekiq.org/) ([caveats](https://github.com/actionhero/node-resque/issues/311)). We also implement some of the popular Resque plugins, including [resque-scheduler](https://github.com/resque/resque-scheduler) and [resque-retry](https://github.com/lantins/resque-retry)
 
-The full API documentation for this package is automatically generated from the `master` via [typedoc](https://typedoc.org) branch and published to https://node-resque.actionherojs.com/
+The full API documentation for this package is automatically generated from the `main` via [typedoc](https://typedoc.org) branch and published to https://node-resque.actionherojs.com/
 
 [![Nodei stats](https://nodei.co/npm/node-resque.png?downloads=true)](https://npmjs.org/package/node-resque)
 
 [![CircleCI](https://circleci.com/gh/actionhero/node-resque.svg?style=svg)](https://circleci.com/gh/actionhero/node-resque)
 
+## The Resque Factory (How It Works)
+### Overview
+Resque is a queue based task processing system that can be thought of as a "Kanban" style factory. Workers in this factory can each only work one Job at a time. They pull Jobs from Queues and work them to completion (or failure). Each Job has two parts: instructions on how to complete the job (the `perform` function), and any inputs necessary to complete the Job.
+
+### Queues
+In our factory example, Queues are analogous to conveyor belts. Jobs are placed on the belts (Queues) and are held in order waiting for a Worker to pick them up. There are two "special" Queues: the Delayed Job Queue, and the Failed Job Queue. The Delayed Job Queue contains Job definitions that are intended to be worked at or in a specified time. The Failed Job Queue is where Workers place any Jobs that have failed during execution.
```

Comment:
There are technically likes of delayed queues, one fro each timestamp in the futute:

* high 
* medium
* low
* delayed_1000
* delayed_2000

At some point after 1000 (timestamp) has passed, it's jobs will be moved into the normal high, medium, or low queues

Reply:
I see. That feels like an implementation detail that might not need to be covered in a high-level description. I can add a bullet point about it down in the more technical description. Or do you think it belongs here instead?

Reply:
No, I don't think we need this much detail here.  Perhaps you rephrase the following:

> There are three types of Queues: regular work Queues, Delayed Job Queues, and the Failed Job Queue. The Delayed Job Queues contains Job definitions that are intended to be worked at or in a specified time. The Failed Job Queue is where Workers place any Jobs that have failed during execution.

Reply:
Something is not clear for me : 
- high
- medium
- low

is just the name of the queue right?  And it's high because it's the first queue in the list ? Only that no?
There is a confusion for me between priority, and the name of the queue provide by this example.

Reply:
Yes - the "high" queue would only be the highet priority if first in the array, eg: `queues = ["high", "medium", "low"]`